### DISCUSSION TITLE
Difference in buffer size for PipeReader

### DISCUSSION THREAD

**Discussion by mbendtsen** [dotnet/runtime#79850] [Q&A]
Considering this code (notice `requestBodyInBytes.Buffer.FirstSpan`):

```csharp
private static async Task<string> GetBody(HttpRequest httpRequest)
{
    ReadResult requestBodyInBytes = await httpRequest.BodyReader.ReadAsync();
    httpRequest.BodyReader.AdvanceTo(requestBodyInBytes.Buffer.Start, requestBodyInBytes.Buffer.End);
    return Encoding.UTF8.GetString(requestBodyInBytes.Buffer.FirstSpan);
}
```
When running local (windows) this works fine with a body around 1.500 bytes. But in our production environment (aks, linux) the body would be cut off around 499 bytes. 

Is there a difference in size of the buffer for different environment? (not quite sure how the PipeReader works beneath the surface)

To fix my issue though I just used the whole buffer:
```csharp
private static async Task<string> GetBody(HttpRequest httpRequest)
{
    ReadResult requestBodyInBytes = await httpRequest.BodyReader.ReadAsync();
    httpRequest.BodyReader.AdvanceTo(requestBodyInBytes.Buffer.Start, requestBodyInBytes.Buffer.End);
    return  Encoding.UTF8.GetString(requestBodyInBytes.Buffer);
}
```

**Comment by davidfowl** [comment]
Does this help https://www.stevejgordon.co.uk/an-introduction-to-sequencereader?
